shaal/DrupalPod
# Adding Drupal projects through composer

# The Problem/Issue/Bug
Modules/Themes source code is added manually, without all the required dependencies

## How this PR Solves The Problem
This PR use composer to require the project so all the dependencies are downloaded as well.
It uses the git repository source code of the project (ie. `/repo/upgrade_status`), that is symlinked into the expected location in a Drupal project (ie. `/web/modules/contrib/upgrade_status`).

## Manual Testing Instructions
This link will open upgrade_status project using this Repo
https://gitpod.io/#DP_PROJECT_NAME=upgrade_status,DP_ISSUE_FORK=,DP_ISSUE_BRANCH=,DP_PROJECT_TYPE=project_module,DP_MODULE_VERSION=8.x-3.x,DP_CORE_VERSION=8.9.x,DP_PATCH_FILE=https%3A%2F%2Fwww.drupal.org%2Ffiles%2Fissues%2F2021-07-21%2F3224660_1.patch,DP_INSTALL_PROFILE=standard/https://github.com/shaal/DrupalPod/tree/module-dependencies

Otherwise, you can use the DrupalPod browser extension, but you should go to its options, and change the repo to `https://github.com/shaal/DrupalPod/tree/module-dependencies`
After that change, go to any Drupal issue page, and use the extension as usual. It will open with the changes of this PR.

## Related Issue Link(s)
This PR was created, after `preferred-source` experiment failed due to limitations - https://github.com/shaal/DrupalPod/pull/25

## Release/Deployment notes



<a href="https://gitpod.io/#https://github.com/shaal/DrupalPod/pull/26"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
